### PR TITLE
Release Candidate 2.21.0

### PR DESCRIPTION
**Pull request by jakebrinkmann**
## Status
**READY FOR SYSTEM TEST**

## Description
Updates required for hadoop-2.8.0 migration

## Related PRs

branch | PR
------ | ------
2017.07.Dev | [espa-api#89](https://github.com/USGS-EROS/espa-api/pull/89)

## Deploy Notes

* These changes assume [hadoop-2.8.0](http://hadoop.apache.org/docs/r2.8.0/index.html) be installed/linked to `~/bin/hadoop`
  * Cluster-specific configuration assumed occurs in `~/bin/hadoop/etc/hadoop/`

## Impacted Areas

* Hadoop 2.8.0 Map/Reduce Job Submission 
  + Update deprecated `hadoop` calls using  `hdfs`/`yarn` equivalent commands
  + Use the `-files` argument, to stage the entire core application as a directory
  + Remove unused environment variables
  + Job names formatted as timestamp `%Y-%m-%d-%H-%M-%S`
* Bug fix to allow pixel_qa from OLI-Only data

### CODE REVIEW

File: scheduling/ondemand_cron.py [hunk 1]
```diff
@@ -148,12 +148,24 @@ def process_requests(cron_cfg, proc_cfg, args,
...
     # are over a limit
     job_limit = cron_cfg.getint('hadoop', 'max_jobs')
-    cmd = "hadoop job -list|awk '{print $1}'|grep -c job 2>/dev/null"
+    yarn_running_apps_command = [yarn_executable, "application",
+                                 "-appStates", "RUNNING", "-list"]
     try:
-        job_count = execute_cmd(cmd)
+        cmd = ' '.join(yarn_running_apps_command)
+        app_states = execute_cmd(cmd)
+        job_count = [l for l in app_states.split('\n')
```
Comment (ronaldddilley) [inline scheduling/ondemand_cron.py]: Could you add a comment here.  To explain you are only looking for one line of the output.

File: processing/processor.py [hunk 2]
```diff
@@ -1426,11 +1426,6 @@ def validate_parameters(self):
             raise Exception('include_cfmask is an unavailable product option'
                             ' for OLI-Only data')
 
-        if options['include_pixel_qa'] is True:
```
Comment (ronaldddilley) [inline processing/processor.py]: Why are we removing this check?

File: README.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ This project contains the code for processing a single input dataset to the fina
 Please see the Release Tags for notes related to past versions.
 
 - Version change for system wide versioning
+- Commandline changes for Hadoop 2.8.0
```
Comment (ronaldddilley) [inline README.md]: May need to add a comment for removing the pixel_qa check in processor.py.

File: scheduling/ondemand_cron.py (hunk 1 above)
Comment (jakebrinkmann) [inline scheduling/ondemand_cron.py]: Added comment in 34e984a

File: README.md (hunk 3 above)
Comment (jakebrinkmann) [inline README.md]: Added in 8aed3e9

File: processing/processor.py (hunk 2 above)
Comment (jakebrinkmann) [inline processing/processor.py]: verified outputs completed match oli/tirs pixel_qa only

**Pull request merged**